include/envoy/server/hot_restart.h

```diff
@@ -11,6 +13,16 @@ namespace Server {
 
 class Instance;
 
+class OsSysCalls {
```

Comment:
Do you want to move this to its own file in somewhere more neutral than hotrestart? I think this would be a useful shim layer for syscalls to improve mocking and testing across the code base.

Reply:
I took the idea for this from Envoy::Filesystem::OsSysCalls.  I can see the utility of both having one big OsSysCalls shim for all of envoy, and for having a smaller one per use.  I'm fine with either; we should just decide which one we like better.  @mattklein123 what's your opinion?

Reply:
It sounds good to me to have a single OS shim layer. Perhaps move into the Api namespace?

Reply:
Want me to do that in this PR or open a separate one?

Reply:
Either way. This PR is fine if it's easier for you.

Reply:
Ok, let's combine the OsSysCalls in a separate PR then.  Keep the intent/scope of this one clean.

Reply:
See #1791